src/PowerShellEditorServices/Session/Host/TerminalPSHostRawUserInterface.cs

```diff
@@ -184,11 +145,87 @@ public override Size MaxWindowSize
         /// <returns>A KeyInfo struct with details about the current keypress.</returns>
         public override KeyInfo ReadKey(ReadKeyOptions options)
```

Comment:
@SteveL-MSFT @daxian-dbw would you mind taking a look at this `ReadKey` implementation if you have a sec? This will be the implementation for `$Host.UI.RawUI.ReadKey()` in the Integrated Console.

I think it looks ok.